initial addition of toString to Query Classes

khayashi opened the pull request:
added to the minimum set of classes Zhouchuan uses.
This is again temporally changes. will redesign whole query classes.

khayashi committed:
initial addition of toString
added to the minimum set of classes Zhouchuan uses.
This is again temporally changes. will redesign whole query classes.

santip commented on clients/java/src/main/java/com/senseidb/search/client/req/query/DisMax.java:
```diff
@@ -81,5 +81,22 @@ public double getBoost() {
+        
+        buffer.append("^");
+        buffer.append(boost);
+        buffer.append("tieBraker:");
```
this will be adjacent to the boost

santip commented on clients/java/src/main/java/com/senseidb/search/client/req/query/TextQuery.java:
```diff
@@ -155,5 +155,15 @@ public Type getType() {
+    public String toString() {
+        StringBuffer buffer = new StringBuffer();
+        buffer.append(operator == null ? "" : operator.toString());
+        buffer.append(" " + super.toString()); // for field
```
FieldAwareQuery could return empty string, consider moving the colon to the super.toString(), only when not empty.

santip commented on clients/java/src/main/java/com/senseidb/search/client/req/query/TextQuery.java:
```diff
@@ -155,5 +155,15 @@ public Type getType() {
+    
+    @Override
+    public String toString() {
+        StringBuffer buffer = new StringBuffer();
```
You can use StringBuilder which is more lightweight as it doesn't synchronize calls.

santip commented on clients/java/src/main/java/com/senseidb/search/client/req/Term.java:
```diff
@@ -55,4 +55,15 @@ public String getValue() {
+    
+    @Override
+    public String toString() {
+        StringBuffer buffer = new StringBuffer();
```
Use StringBuilder

santip commented on clients/java/src/main/java/com/senseidb/search/client/req/Term.java:
```diff
@@ -55,4 +55,15 @@ public String getValue() {
+    public String toString() {
+        StringBuffer buffer = new StringBuffer();
+        buffer.append(super.toString()); // for field
+        buffer.append(":");
```
super.toString() can be empty, move the colon there

santip commented on clients/java/src/main/java/com/senseidb/search/client/req/Operator.java:
```diff
@@ -21,4 +21,6 @@
 public enum Operator {
     and, or;
+    
+    
```
Please remove this additional whitespace

santip commented on clients/java/src/main/java/com/senseidb/search/client/req/query/BoolQuery.java:
```diff
@@ -80,5 +80,46 @@ public double getBoost() {
+    @Override 
+    public String toString() {
+        StringBuffer buffer = new StringBuffer();
+        boolean needParens=(getBoost() != 1.0) || (getMinimumNumberShouldMatch() > 0) ;
```
You're not using these conditionals to avoid appending the boost info or the minimum should match.

santip commented:
Change all StringBuffers for StringBuilders

rahula commented on clients/java/src/main/java/com/senseidb/search/client/req/query/BoolQuery.java:
```diff
@@ -80,5 +80,46 @@ public double getBoost() {
+          buffer.append("(");
+        }
+        
+        for (Query q : must) {
```
you might want to rewrite it on a single line to improve the readability 
buffer.append("+(").append.(q).append(")")

khayashi committed:
Use StringBuilder.
regardless of its value, always print boost.since it might have a bug in
setting boost.

chtyim commented on clients/java/src/main/java/com/senseidb/search/client/req/query/DisMax.java:
```diff
@@ -81,5 +81,17 @@ public double getBoost() {
+    
+    @Override
+    public String toString() {
+        StringBuffer buffer = new StringBuffer();
```
Use StringBuilder

khayashi committed:
StringBuilder in DisMax.
Forgot to fix it in previous commit
